src/pkg/packaging/windows/vscomponents/generatenupkg.ps1

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+param(
+    [Parameter(Mandatory=$true)][string]$MsiPath,
+    [Parameter(Mandatory=$true)][string]$MsiVersion,
+    [Parameter(Mandatory=$true)][string]$NuspecFile,
+    [Parameter(Mandatory=$true)][string]$NupkgFile,
+    [Parameter(Mandatory=$true)][string]$Architecture,
+    [Parameter(Mandatory=$true)][string]$ComponentName,
+    [Parameter(Mandatory=$true)][string]$ComponentFriendlyName,
+    [Parameter(Mandatory=$true)][string]$ProjectUrl
+)
+
+$RepoRoot = Convert-Path "$PSScriptRoot\..\..\..\..\..\.."
+$NuGetDir = Join-Path $RepoRoot "artifacts\Tools\nuget"
+$NuGetExe = Join-Path $NuGetDir "nuget.exe"
+$OutputDirectory = [System.IO.Path]::GetDirectoryName($NupkgFile)
+$SdkBundlePath = [System.IO.Path]::GetFullPath($MsiPath)
+
+if (-not (Test-Path $NuGetDir)) {
+    New-Item -ItemType Directory -Force -Path $NuGetDir | Out-Null
+}
+
+if (-not (Test-Path $NuGetExe)) {
+    # Using 3.5.0 to workaround https://github.com/NuGet/Home/issues/5016
+    Write-Output "Downloading nuget.exe to $NuGetExe"
+    wget https://dist.nuget.org/win-x86-commandline/v3.5.0/nuget.exe -OutFile $NuGetExe
+}
+
+if (Test-Path $NupkgFile) {
+    Remove-Item -Force $NupkgFile
+}
+
+& $NuGetExe pack $NuspecFile -Version $MsiVersion -OutputDirectory $OutputDirectory -NoDefaultExcludes -NoPackageAnalysis -Properties COMPONENT_MSI=$MsiPath`;ARCH=$Architecture`;COMPONENT_NAME=$ComponentName`;FRIENDLY_NAME=$ComponentFriendlyName`;PROJECT_URL=$ProjectUrl
```

Comment:
The version here is in what format? In AspNetCore, I'm passing in the shareframework version, which looks like `3.0.0-preview7`. This slightly differs from what you have stated as the package names above.

Reply:
The '$MsiVersion' should be the 'product' or 'build' version. Example: "3.0.0-preview7-27821-01"